shell/platform/android/io/flutter/embedding/FlutterFragment.java

```diff
@@ -0,0 +1,425 @@
+  }
+
+  /**
+   * Hook for subclasses to customize the creation of the {@code FlutterNativeView}.
+   *
+   * This method is only invoked from the default implementation of {@link #createFlutterView(Activity)}.
+   * If {@link #createFlutterView(Activity)} is overridden, then this method will not be invoked unless
+   * it is invoked directly from the subclass.
+   *
+   * By default, this method returns a standard {@link FlutterNativeView} without any modification.
+   */
+  @NonNull
+  protected FlutterNativeView createFlutterNativeView(@NonNull Context context) {
+    return new FlutterNativeView(context);
+  }
+
+  /**
+   * Hook for subclasses to customize aspects of the {@code FlutterView} that are not creation
+   * dependent, e.g., {@code FlutterView}'s initial route.
+   */
+  protected void onFlutterViewCreated(@SuppressWarnings("unused") @NonNull FlutterView flutterView) {
+    // no-op
+  }
+
+  /**
+   * Creates a {@link View} containing the same {@link Drawable} as the one set as the
+   * {@code windowBackground} of the parent activity for use as a launch splash view.
+   *
+   * Shows and then automatically animates out the launch view when Flutter renders its first frame.
+   *
+   * {@code FlutterView} must exist before this method is called.
+   *
+   * Returns null if no {@code windowBackground} is set for the activity.
+   */
+  @SuppressLint("NewApi")
+  @Nullable
+  private View createLaunchView() {
+    if (!isSplashScreenDesired()) {
+      return null;
+    }
+
+    final Drawable launchScreenDrawable = getSplashScreenDrawableFromActivityTheme();
+    if (launchScreenDrawable == null) {
+      return null;
+    }
+
+    final View view = new View(getContextCompat());
+    view.setLayoutParams(MATCH_PARENT);
+    // TODO(mattcarroll): this API version error exists in the original code. Why is it there?
+    view.setBackground(launchScreenDrawable);
+
+    flutterView.addFirstFrameListener(new FlutterView.FirstFrameListener() {
+      @Override
+      public void onFirstFrame() {
+        launchView.animate()
+            .alpha(0f)
+            // Use Android's default animation duration.
+            .setListener(new AnimatorListenerAdapter() {
+              @Override
+              public void onAnimationEnd(Animator animation) {
+                // Views added to an Activity's addContentView is always added to its
+                // root FrameLayout.
+                ((ViewGroup) launchView.getParent()).removeView(launchView);
+                launchView = null;
+              }
+            });
+
+        flutterView.removeFirstFrameListener(this);
+      }
+    });
+
+    // Resets the activity theme from the one containing the launch screen in the window
+    // background to a blank one since the launch screen is now in a view in front of the
+    // FlutterView.
+    //
+    // We can make this configurable if users want it.
+    // TODO: get rid of Activity references.
```

Comment:
What is the proper TODO structure so this does not get lost? Should you be filing issues and then linking them here?

Reply:
I don't think we need to file issues at this juncture because the work that is already ticketed will resolve most/all of these TODOs.  But I should have placed my ldap in the todo.